File: templates/problems/07_trees/05_same_tree/same_tree_test.go
package same_tree

import "testing"

func TestIsSameTree(t *testing.T) {
	tests := []struct {
		name string
		p    []int
		q    []int
		want bool
	}{
		{"identical", []int{1, 2, 3}, []int{1, 2, 3}, true},
		{"different structure", []int{1, 2}, []int{1, -101, 2}, false},
		{"different values", []int{1, 2, 1}, []int{1, 1, 2}, false},
		{"both empty", []int{}, []int{}, true},
		{"one empty", []int{1}, []int{}, false},
		{"single same", []int{1}, []int{1}, true},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			p := buildTreeHelper(tt.p)
			q := buildTreeHelper(tt.q)
			if got := isSameTree(p, q); got != tt.want {
				t.Errorf("isSameTree() = %v, want %v", got, tt.want)
			}
		})
	}
}

File: templates/problems/07_trees/05_same_tree/typescript/same_tree.test.ts
import { describe, it, expect } from "vitest";
import { buildTree } from "./helpers";
import { isSameTree } from "./same_tree";

describe("isSameTree", () => {
    it.each([
        { name: "identical", p: [1, 2, 3], q: [1, 2, 3], want: true },
        { name: "different structure", p: [1, 2], q: [1, -101, 2], want: false },
        { name: "different values", p: [1, 2, 1], q: [1, 1, 2], want: false },
        { name: "both empty", p: [], q: [], want: true },
        { name: "one empty", p: [1], q: [], want: false },
        { name: "single same", p: [1], q: [1], want: true },
    ])("case: $name", ({ p, q, want }) => {
        expect(isSameTree(buildTree(p), buildTree(q))).toBe(want);
    });
});
